Modules/_elementtree.c

```diff
@@ -1449,6 +1449,23 @@ element_getitem(PyObject* self_, Py_ssize_t index)
     return Py_NewRef(self->extra->children[index]);
 }
 
+static int
+element_bool(PyObject* self_)
+{
+    ElementObject* self = (ElementObject*) self_;
+    if (PyErr_WarnEx(PyExc_DeprecationWarning,
+                     "DeprecationWarning: The behavior of this method will "
+                     "change in future versions.  Use specific 'len(elem)' or "
+                     "'elem is not None' test instead.",
+                     2) < 0) {
```

Comment:
have you confirmed that this stacklevel puts the warning in the right place on both the C and Python implementations?

Reply:
Thanks. The C implementation was not in the right place; I'd just copied it over from Python. Maybe I'd only checked it in the REPL. I only get the correct results for both `bool(el)` and `if(el)` on the REPL and from a script when using a stacklevel of 2 for Python and 1 for C. (I don't know the reason.)

Reply:
Great, I suspected you might need 1 here. The reason for the difference is that it's only Python stack frames that the warnings code can count. A call into a C function is not a Python frame.